docs/proposal-0005-jetstream-tooling-versioning.md

```diff
@@ -0,0 +1,62 @@
+# Preventing inconsistent analysis results after tooling updates 
+
+Currently, whenever tooling, such as mozanalysis, metric-hub or jetstream gets updated, a new Docker container gets pushed to GCR and is automatically used during the next analysis run. This can potentially cause inconsistent analysis results for experiments that are currently live if the way results are computed changes. This proposal describes how jetstream could decide to use the new versions of tooling only for experiments that launch after updates have been pushed, and keep using older versions of the tooling based on the tooling versions the analyses initially started with.
+
+## Keeping track of library and framework versions
+
+Jetstream is based on mozanalysis, metric-config-parser and some internal logic. When a new version of jetstream is released, a new Docker container gets pushed to [GCR](https://console.cloud.google.com/gcr/images/moz-fx-data-experiments/global/jetstream?project=moz-fx-data-experiments). The container installs a specific version of each library and can be uniquely identified by a SHA256 hash. A timestamp indicating when the container was uploaded is also available.
+
+Every time Jetstream runs and writes computed results to BigQuery, it tags the result tables with a last updated timestamp. 
```

Comment:
This line threw me off at first. I was confused about using the results tables' last updated timestamp and how that would give us a consistent version of the image. This is probably just a "me" problem, but I don't think it hurts to add another sentence or two to clarify?

```suggestion
Every time Jetstream runs and writes computed results to BigQuery, it tags the result tables with a last updated timestamp. However, the enrollments table won't update on new Jetstream runs, giving us an anchor from which to identify a consistent Jetstream image hash.
```

Reply:
Good point!